src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -721,7 +719,23 @@ private static bool IsArrayItemContext(List<SyntaxBase> matchingNodes, (ArraySyn
 
                 return new(
                     Function: function,
-                    ArgumentIndex: previousArg is null ? 0 : (args.IndexOf(previousArg) + 1));
+                    ArgumentIndex: previousArg is null ? 0 : (args.IndexOf(previousArg) + 1),
+                    ArgumentNodes: ImmutableArray<SyntaxBase>.Empty);
+            }
+
+            // someFunc(x|yz) or someFunc(xy|z) or someFunc(xyz|)
+            // abc.someFunc(x|yz) or abc.someFunc(xy|z) or abc.someFunc(xyz|)
+            var (argumentNode, argumentIndex) = SyntaxMatcher.FindLastNodeOfType<FunctionArgumentSyntax, FunctionArgumentSyntax>(matchingNodes);
+            if (argumentNode is not null)
```

Comment:
I think we should stick to the existing pattern and be precise here - if you're just trying to match a string parameter, should be able to can do:
```csharp

if (SyntaxMatcher.IsTailMatch<FunctionCallSyntaxBase, FunctionArgumentSyntax, StringSyntax, Token>(matchingNodes, (_, _, _, _) => true))
{
    var function = (FunctionCallSyntaxBase)matchingNodes[^4];
    var args = function.Arguments.ToImmutableArray();
    var argIndex = args.IndexOf((FunctionArgumentSyntax)matchingNodes[^3]);
    ...
```

Reply:
Not only, I'd like to show file completions even if you have typed a symbol that's name would also match a file in your directory.

Reply:
Additionally, this is only for creating FunctionArgumentContext. the kind still remains as it was before - when there are exact matches.